arrays/arrays.c

```diff
@@ -169,13 +222,45 @@ void arr_append(Array *arr, char *element)
  *****/
 void arr_remove(Array *arr, char *element)
 {
+    printf("~~arr_remove~~\n");
 
     // Search for the first occurence of the element and remove it.
     // Don't forget to free its memory!
+    int lastElement = arr->count - 1;
+    // Saving for after the search loop so we know when to stop shifting elements down
+    int index = 0;
+    for (; index < arr->count; index++)
+    {
+        if (*arr->elements[index] == *element)
+        {
+            free(arr->elements[index]);
+            break;
+        }
+
+        // In case the passed element isn't in the array...
+        if (&arr->elements[index] == &arr->elements[lastElement] && arr->elements[index] != arr->elements[lastElement])
```

Comment:
jaja nice, but would simply `index == lastElement` not work for the first half of your if statement?